src/components/FallenHeroes.tsx

```diff
@@ -1,54 +1,63 @@
-import FallenHero from "./FallenHero"
-import { allHerosMock, state } from "../context";
-import { persistData } from "../context";
-import React from 'react'
-
-function FallenHeroes(){
-    // the FallenHero component should be rendered in this component in a loop or so
-    // to show all the fallen heroes, via forEach() or so
-      
-    // addFallenHero() - unfinished function for the send to coffin button
-    // to get the hero from the input text and if correct, that is
-    // matching hero from the set of totalHeroes,
-    // and is included in rest of heroes alive, update the state accordingly
-    const addFallenHero = function(name:string):void{
-        //add hero to the list if it isnt there yet
-        if(!state.restOfAliveHeroes.includes(name)){
-            throw Error
-        } 
-        if(allHerosMock.includes(name) && state.restOfAliveHeroes.includes(name)){
-            state.fallenHeroes.push(name);
-            delete state.restOfAliveHeroes[name]
-            persistData();
-            console.log(state)
-        }
-        if(!allHerosMock.includes(name)){
-            throw Error
-        } 
-
+import { LotrContext } from "../context";
+import React, { useState } from "react";
+
+function FallenHeroes() {
+  // the FallenHero component should be rendered in this component in a loop or so
+  // to show all the fallen heroes, via forEach() or so
+
+  // addFallenHero() - unfinished function for the send to coffin button
+  // to get the hero from the input text and if correct, that is
+  // matching hero from the set of totalHeroes,
+  // and is included in rest of heroes alive, update the state accordingly
+  const { campaign, setCampaign } = React.useContext(LotrContext);
+  const [fallenHero, setFallenHero] = useState("");
+
+  const handleChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
+    setFallenHero(e.target.value);
+  };
+
+  function addFallenHero() {
```

Comment:
This is a function declaration, while the above function uses an expression. Choose one style and stick to it.

Reply:
unified